Source/WebCore/platform/graphics/skia/GraphicsContextSkia.h

```diff
@@ -113,6 +113,15 @@ class WEBCORE_EXPORT GraphicsContextSkia final : public GraphicsContext {
             SkPaint::Cap cap { SkPaint::kButt_Cap };
             SkPaint::Join join { SkPaint::kMiter_Join };
         } m_stroke;
+
+        class DashState {
+        public:
+            DashState(DashArray dashArray, float dashOffset) : dashArray(dashArray), dashOffset(dashOffset) { };
+
+            DashArray dashArray;
+            float dashOffset;
+        };
+        std::optional<DashState> m_dash { std::nullopt };
```

Comment:
If there's no array isn't doesn't that mean to not apply a dash? Seems simpler to use that rather than an optional.

Reply:
I've moved the `std::optional` to the `DashArray` now, and the array is used to apply the dash effect now. Is that what you were suggesting here?

Reply:
I was suggesting that you check `DashArray`'s size rather than using `std::optional`.

Reply:
Thanks for clarifying that. That works too. Will push in a second.